txpipe/twopoint_fourier.py

```diff
@@ -602,7 +603,11 @@ def compute_noise(self, i, j, k, ell_bins, maps, workspace):
         if k == POS_POS:
             mean_noise /= 4
 
-        return mean_noise
+        if self.config['true_shear']:
```

Comment:
Can you do this test at the start of the function and avoid the calculation if it's not needed?